source/jormungandr/tests/instant_system_new_default_routing_tests.py

```diff
@@ -0,0 +1,223 @@
+}
+
+QUERY_DATETIME_STR = "20120614T070000"
+
+
+def mock_instance_system(_, params):
+    return MockResponse(INSTANCE_SYSTEM_RESPONSE, 200)
+
+
+@pytest.fixture(scope="function", autouse=True)
+def mock_http_instance_system(monkeypatch):
+    monkeypatch.setattr('jormungandr.scenarios.ridesharing.instant_system.InstantSystem._call_service',
+                        mock_instance_system)
+
+
+@dataset({'main_routing_test': MOCKED_INSTANCE_CONF})
+class TestInstanceSystem(NewDefaultScenarioAbstractTestFixture):
+    """
+    Integration test with Instace System
+    """
+
+    def test_basic_ride_sharing(self):
+        """
+        test ridesharing_jouneys details
+        """
+        q = "journeys?from=0.0000898312;0.0000898312&to=0.00188646;0.00071865&datetime=20120614T075500&"\
+            "first_section_mode[]={first}&last_section_mode[]={last}&debug=true"\
+            .format(first='ridesharing', last='walking')
+        response = self.query_region(q)
+        self.is_valid_journey_response(response, q, check_journey_links=False)
+
+        journeys = get_not_null(response, 'journeys')
+        assert len(journeys) == 2
+        tickets = response.get('tickets')
+        assert len(tickets) == 1
+        assert tickets[0].get('cost').get('currency') == 'centime'
+        assert tickets[0].get('cost').get('value') == '170.0'
+        ticket_id = tickets[0].get('id')
+
+        ridesharing_kraken = journeys[0]
+        assert 'ridesharing' in ridesharing_kraken['tags']
+        assert 'non_pt' in ridesharing_kraken['tags']
+        assert ridesharing_kraken.get('type') == 'best'
+        assert ridesharing_kraken.get('durations').get('ridesharing') == 13
+        assert ridesharing_kraken.get('durations').get('total') == 13
+
+        rs_sections = ridesharing_kraken.get('sections')
+        assert len(rs_sections) == 1
+        assert rs_sections[0].get('mode') == 'ridesharing'
+        assert rs_sections[0].get('type') == 'street_network'
+
+        ridesharing = journeys[1]
+        assert 'ridesharing' in ridesharing['tags']
+        assert 'ecologic' in ridesharing['tags']
+        assert ridesharing.get('type') == 'rapid'
+
+        distances = ridesharing.get('distances')
+        assert distances.get('ridesharing') == 211
+        assert distances.get('car') == 0
+        assert distances.get('walking') == 0
+
+        durations = ridesharing.get('durations')
+        assert durations.get('ridesharing') == 926
+        assert durations.get('total') == 926
+
+        rs_sections = ridesharing.get('sections')
+        assert len(rs_sections) == 1
+        assert rs_sections[0].get('mode') == 'ridesharing'
+        assert rs_sections[0].get('type') == 'crow_fly'
+
+        rs_journeys = rs_sections[0].get('ridesharing_journeys')
+        assert len(rs_journeys) == 1
+        assert rs_journeys[0].get('distances').get('walking') == 16
```

Comment:
not sure it's mandatory to test that (we pretty much don't care if it is tracked or not)